lib/GitterGhost.js

```diff
@@ -27,7 +49,44 @@ GitterGhost.prototype._getRoom = function(roomName) {
     });
 };
 
+GitterGhost.prototype._couldBeBanned = function(text) {
+    /*
+        Gitter bans users for sending too many messages with the same has in a set period (12 hours).
+        It's rather impractical, but the best way to deal with this is to pre-empt them and check the 
+        thesholds ourselves by storing all the hashes locally. Gitter also do not make any distinction
+        about the destination room where the message may be sent, so we must apply these limits globally
+        which is a bit lame.
+     */
+    const hash = crypto.createHash('md5').update(text).digest('hex');
+    // Lifted from https://gitlab.com/gitlab-org/gitter/webapp/blob/develop/modules/spam-detection/lib/duplicate-chat-detector.js#L26
```

Comment:
I think we have to do something regarding licensing - would invoke a Matthew as the legal expert (or person to redirect you to a legal expert).

It's currently listed as MIT, which has a clause about including text about warranties and things.